From 7f0f47269451f0dfcb5e256ccffe71b359248087 Mon Sep 17 00:00:00 2001
From: Igal Liberman <igall@marvell.com>
Date: Tue, 6 Jun 2017 18:07:03 +0300
Subject: [PATCH 1026/1345] dts: ap806: add reserved memory for PSCI area

commit  75fc80d523f4c852111f71927afe97cd334930ff from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

The PSCI area should be reserved in Linux for PSCI operations.
This patch reserves 2M byte memory for PSCI, which covers all PSCI
code and data area, and be 2M byte aligned which is required by
Linux 2M byte big pages.

This patch fixes possible hangs when Linux uses PSCI code,
for example:
	- Power management operations
	- Using reboot command

Change-Id: Ia080c28a10a01ab319793e0c83dd15e117bed8f7
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/40234
Reviewed-by: Omri Itach <omrii@marvell.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/40424
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index 23f0d65..8e2309c 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -66,6 +66,17 @@
 		method = "smc";
 	};
 
+	reserved-memory {
+		#address-cells = <2>;
+		#size-cells = <2>;
+		ranges;
+
+		psci-area@4000000 {
+			reg = <0 0x4000000 0 0x200000>;
+			no-map;
+		};
+	};
+
 	/* CPUfreq Cluster settings:
 	 * - Initial required static OPP entry of 100Mhz - suited for all CPU boot options
 	 * - Additional OPP entries are registered at runtime by cpufreq driver
-- 
1.7.9.5

